### PR TITLE
Add npath tool

### PR DESCRIPTION
**Pull request by dpaz**
The test files are yet to be added to the PR, I don't know where they should be.
This PR depends on [#122](https://github.com/bblfsh/sdk/pull/122)

### CODE REVIEW

**Review by alcortesm — CHANGES_REQUESTED** [review]
I guess `NpathComplexity` has to be tested on a file called `npath_test.go`

**Comment by dpaz** [comment]
yes, the .go file is ok, the problem is that I have some extra files which I use to do some tests. That's are the files that I have no idea where they should be

**Comment by smola** [comment]
@dpaz Let's add these extra files to a directory `fixtures`.

**Review by abeaumont — COMMENTED** [review]
I think it makes sense to have proper extension in fixture files, since we may/should have variants in different programming languages

**Review by alcortesm — APPROVED** [review]

File: npath.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+import (
+	"fmt"
+
+	"github.com/bblfsh/sdk/uast"
+)
+
+//CodeReference
+//https://pmd.github.io/pmd-5.7.0/pmd-java/xref/net/sourceforge/pmd/lang/java/rule/codesize/NPathComplexityRule.html
+
```
Comment (smola) [inline npath.go]: Use an appropiate godoc comment here. For example:

```
Npath computes the NPath of functions in a *uast.Node.

PMD is considered the reference implementation to assert correctness.
See: https://pmd.github.io/pmd-5.7.0/pmd-java/xref/net/sourceforge/pmd/lang/java/rule/codesize/NPathComplexityRule.html
```

Also, rename `Npath` to `NPath` everywhere.

**Review by smola — CHANGES_REQUESTED** [review]

**Comment by smola** [comment]
@dpaz This needs a rebase to fix conflicts with the cyclomatic complexity PR, which is now merged.

**Review by abeaumont — APPROVED** [review]

**Review by smola — APPROVED** [review]

**Comment by smola** [comment]
@dpaz can you rebase and squash all superfluous commits?

**Comment by dpaz** [comment]
@smola done

**Pull request merged**